src/txn/txn_timestamp.c

```diff
@@ -671,28 +677,34 @@ __wt_txn_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
 	WT_DECL_RET;
 	WT_TXN *txn;
 	wt_timestamp_t ts;
-	bool prepare, prepare_allowed;
+	bool prepare, prepare_allowed, round_to_prepare;
 
 	txn = &session->txn;
+	prepare = prepare_allowed = round_to_prepare = false;
 	prepare = F_ISSET(txn, WT_TXN_PREPARE);
-	prepare_allowed = prepare == false;
```

Comment:
Was the setting of prepare_allowed here wrong or redundant before?

Reply:
the mistake in this statement is prepare is overwritten, which is fetched in the above statement.